File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Concepts/ClientAPI/ExternalAPI/WikipediaIntegrationPage.vb
Imports Gizmox.WebGUI.Forms.Client

Namespace CompanionKit.Concepts.ClientAPI.ExternalAPI

    Public Class WikipediaIntegrationPage

        Public Sub New()

            ' This call is required by the Visual WebGui UserControl Designer.
            InitializeComponent()



        End Sub

        ''' <summary>
        ''' Handles the ClientEnterKeyDown event of the mobjQueryTextBox control.
        ''' </summary>
        ''' <param name="objSender">The source of the event.</param>
        ''' <param name="objArgs">The <see cref="Gizmox.WebGUI.Forms.Client.ClientEventArgs"/> instance containing the event data.</param>
        Private Sub mobjQueryTextBox_ClientEnterKeyDown(objSender As System.Object, objArgs As Gizmox.WebGUI.Forms.Client.ClientEventArgs) Handles mobjQueryTextBox.ClientEnterKeyDown
            SearchWikipedia(objArgs.Context)
        End Sub

        ''' <summary>
        ''' Handles the ClientClick event of the mobjSearchButton control.
        ''' </summary>
        ''' <param name="objSender">The source of the event.</param>
        ''' <param name="objArgs">The <see cref="Gizmox.WebGUI.Forms.Client.ClientEventArgs"/> instance containing the event data.</param>
        Private Sub mobjSearchButton_ClientClick(objSender As System.Object, objArgs As Gizmox.WebGUI.Forms.Client.ClientEventArgs) Handles mobjSearchButton.ClientClick
            SearchWikipedia(objArgs.Context)
        End Sub

        ''' <summary>
        ''' Searches the wikipedia.
        ''' </summary>
        ''' <param name="objClientContext">The obj client context.</param>
        Private Sub SearchWikipedia(objClientContext As ClientContext)
            objClientContext.Invoke("invokeSearchhWikiPedia", mobjQueryTextBox.ID, mobjResultHtmlBox.ID)
        End Sub
    End Class

End Namespace

File: Sources 4.5.2/Gizmox.WebGUI.Forms.CompanionKit/Concepts/ClientAPI/ExternalAPI/WikipediaIntegrationPage.cs
using System;
using System.Text;
using System.Data;
using System.Drawing;
using System.ComponentModel;
using System.Collections.Generic;

using Gizmox.WebGUI.Forms;
using Gizmox.WebGUI.Common;
using Gizmox.WebGUI.Forms.Client;

namespace CompanionKit.Concepts.ClientAPI.ExternalAPI
{
    public partial class WikipediaIntegrationPage : UserControl
    {
        public WikipediaIntegrationPage()
        {
            InitializeComponent();
        }

        /// <summary>
        /// Handles the ClientEnterKeyDown event of the mobjQueryTextBox control.
        /// </summary>
        /// <param name="objSender">The source of the event.</param>
        /// <param name="objArgs">The <see cref="Gizmox.WebGUI.Forms.Client.ClientEventArgs"/> instance containing the event data.</param>
        private void mobjQueryTextBox_ClientEnterKeyDown(object objSender, ClientEventArgs objArgs)
        {
            SearchWikipedia(objArgs.Context);
        }

        /// <summary>
        /// Handles the ClientClick event of the mobjSearchButton control.
        /// </summary>
        /// <param name="objSender">The source of the event.</param>
        /// <param name="objArgs">The <see cref="Gizmox.WebGUI.Forms.Client.ClientEventArgs"/> instance containing the event data.</param>
        private void mobjSearchButton_ClientClick(object objSender, ClientEventArgs objArgs)
        {
            SearchWikipedia(objArgs.Context);
        }

        /// <summary>
        /// Searches the wikipedia.
        /// </summary>
        /// <param name="objClientContext">The obj client context.</param>
        private void SearchWikipedia(ClientContext objClientContext)
        {
            objClientContext.Invoke("invokeSearchhWikiPedia", mobjQueryTextBox.ID, mobjResultHtmlBox.ID);
        }

    }
}